docs/en/authenticators.rst

```diff
@@ -26,6 +26,25 @@ Configuration options:
    identifier in your user storage. Defaults to ``username``. This option is
    used when the ``identify`` option is set to true.
 
+PrimaryKeySession
+=================
+
+This is an improved version of Session that will only store the primary key.
+This way the data will always be fetched fresh from the DB and issues like
+having to update the identity when updating account data should be gone.
+
+It also helps to avoid session invalidation.
+Session itself stores the entity object including nested objects like DateTime or enums.
+With only the ID stored, the invalidation due to objects being modified will also dissolve.
+
+Make sure to match this with a Token identifier with ``key``/``id`` keys:
+
+    $service->loadIdentifier('Authentication.Token', [
+        'tokenField' => 'id', // lookup for DB table
```

Comment:
This feels like it could be risky when combined with other authenticators.

Reply:
Token field? thats the DB lookup field internally, that should be fine
The dataField is the one coming from the authenticator and possibly matching another identifier

Reply:
> This feels like it could be risky when combined with other authenticators.

I have similar concerns. While our default configs might not have any config overlap, we should ensure there isn't a possibility of opening a vulnerability even with custom config and potentially end up with something similar to the algo config vulnerability that were found in various JWT libs.

I think we should first work of #711 to directly associate identifiers with authenticators and add this new authenticator only after that.

Reply:
https://github.com/cakephp/authentication/pull/712 is open
But IMO those don't collide.
The deprecation in parallel together with new docs will make sure people move over to a safer approach.
While it should still work out for everyone so far even if on the deprecated approach.